insights/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,446 @@
+
+const (
+	// GathererStateEnabled gatherer state, which means that the gatherer will run.
+	GathererStateEnabled GathererState = "Enabled"
+	// GathererStateDisabled gatherer state, which means that the gatherer will not run.
+	GathererStateDisabled GathererState = "Disabled"
+)
+
+// dataGatherState declares valid gathering state types
+// +kubebuilder:validation:Optional
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Running' && self == 'Pending')", message="dataGatherState cannot transition from Running to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Pending')", message="dataGatherState cannot transition from Completed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Pending')", message="dataGatherState cannot transition from Failed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Running')", message="dataGatherState cannot transition from Completed to Running"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Running')", message="dataGatherState cannot transition from Failed to Running"
+type DataGatherState string
+
+const (
+	// Data gathering is running
+	DataGatherStateRunning DataGatherState = "Running"
+	// Data gathering is completed
+	DataGatherStateCompleted DataGatherState = "Completed"
+	// Data gathering failed
+	DataGatherStateFailed DataGatherState = "Failed"
+	// Data gathering is pending
+	DataGatherStatePending DataGatherState = "Pending"
+)
+
+// DataGatherStatus contains information relating to the DataGather state.
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.insightsRequestID) || has(self.insightsRequestID))",message="cannot remove insightsRequestID attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.startTime) || has(self.startTime))",message="cannot remove startTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.finishTime) || has(self.finishTime))",message="cannot remove finishTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.dataGatherState) || has(self.dataGatherState))",message="cannot remove dataGatherState attribute from status"
+// +kubebuilder:validation:Optional
+type DataGatherStatus struct {
+	// conditions is an optional field that provides details on the status of the gatherer job.
+	// It may not exceed 100 items and must not contain duplicates.
+	//
+	// The current condition types are DataUploaded, DataRecorded, DataProcessed, RemoteConfigurationNotAvailable, RemoteConfigurationInvalid
+	//
+	// The DataUploaded condition is used to represent whether or not the archive was successfully uploaded for further processing.
+	// When it has a status of True and a reason of HttpStatus200, the archive was successfully uploaded.
```

Comment:
Instead of `HttpStatus200`, what about something like `Succeeded`?

Reply:
Actually I was checking it and it is also dynamic the same way as `UploadFailed` so there could be `HttpStatus202` etc... I will discuss it with the team and I will get back to it here